src/lab4.py

```diff
@@ -122,19 +126,25 @@ def parse(tokens):
         implicit_tags(tok, currently_open)
         if isinstance(tok, Text):
             if tok.text.isspace(): continue
-            node = TextNode(tok.text)
-            currently_open[-1].children.append(node)
+            parent = currently_open[-1]
```

Comment:
You could factor this ilne out of the containing if statement:

if currently_open:
  parent = currently_open[-1]
else
  parent = None

and remove lines 129, 137 and 143-146.